repository/src/test/java/org/alfresco/repo/content/StorageClassTest.java

```diff
@@ -0,0 +1,313 @@
+import org.alfresco.repo.security.authentication.AuthenticationComponent;
+import org.alfresco.repo.transaction.RetryingTransactionHelper;
+import org.alfresco.service.cmr.repository.ChildAssociationRef;
+import org.alfresco.service.cmr.repository.ContentService;
+import org.alfresco.service.cmr.repository.ContentWriter;
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.alfresco.service.cmr.repository.NodeService;
+import org.alfresco.service.cmr.repository.StoreRef;
+import org.alfresco.service.namespace.QName;
+import org.alfresco.service.transaction.TransactionService;
+import org.alfresco.test_category.OwnJVMTestsCategory;
+import org.alfresco.util.BaseSpringTest;
+import org.alfresco.util.GUID;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.mockito.Spy;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import javax.transaction.NotSupportedException;
+import javax.transaction.SystemException;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+@Category(OwnJVMTestsCategory.class)
+@RunWith(SpringRunner.class) public class StorageClassTest extends BaseSpringTest
+{
+        private static final String DEFAULT_SC = "Default1";
+        private static final String DEFAULT_VALUE = "default";
+        private static final String TEST_NAMESPACE = "TestNameSpace";
+        private TransactionService transactionService;
+        private AuthenticationComponent authenticationComponent;
+
+        private NodeService nodeService;
+        private NodeRef rootNode;
+        @Spy ContentStore mockContentStore;
+        ContentService contentService;
+        ContentStore contentStore;
+
+        @Before
+        public void before()
+        {
+                nodeService = (NodeService) applicationContext.getBean("NodeService");
+                transactionService = (TransactionService) applicationContext.getBean("TransactionService");
+                this.authenticationComponent = (AuthenticationComponent) applicationContext.getBean("authenticationComponent");
+                this.contentService = (ContentService) this.applicationContext.getBean("contentService");
+                this.contentStore = (ContentStore) ReflectionTestUtils.getField(contentService, "store");
+
+                FileContentStore fileContentStore = new FileContentStore(applicationContext, this.contentStore.getRootLocation());
+                mockContentStore = spy(fileContentStore);
+
+                when(mockContentStore.getSupportedStorageClasses()).thenReturn(Set.of(DEFAULT_SC, "Azure", "S3"));
+                ReflectionTestUtils.setField(contentService, "store", mockContentStore);
```

Comment:
Why do we set in `Before` of each method a mock content store but we start most of the methods setting an actual content store?

Reply:
Moved all default checks in one test where we need to set the actual content store to test the default behavior.